src/main/java/quinn/Quinn.java

```diff
@@ -0,0 +1,312 @@
+package quinn;
+
+import quinn.exception.QuinnException;
+import quinn.task.Deadline;
+import quinn.task.Event;
+import quinn.task.Task;
+import quinn.task.ToDo;
+import quinn.ui.Ui;
+import quinn.storage.Storage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Quinn {
+    private final Ui ui;
+    private Storage storage;
+    private List<Task> tasks;
+
+    public Quinn(String folderName, String fileName) {
+        ui = new Ui();
+
+        try {
+            storage = new Storage(folderName, fileName);
+            tasks = storage.loadTasksFromFile();
+        } catch (QuinnException | IOException e) {
+            ui.displayError(e.getMessage());
+            tasks = new ArrayList<>();
+        }
+    }
+
+    public static void main(String[] args) {
+        new Quinn("data", "tasks.txt").run();
+    }
+
+    public void run() {
+        ui.displayWelcome();
+
+        String commandLine = "";
+
+        while (!commandLine.equals("bye")) {
+            commandLine = ui.readCommand();
+
+            try {
+                processCommand(commandLine);
+            } catch (QuinnException | IOException e) {
+                ui.displayError(e.getMessage());
+            } finally {
+                ui.displayLine();
+            }
+        }
+    }
+
+    public void processCommand(String commandLine) throws QuinnException, IOException {
```

Comment:
Perhaps it is better to use a more specific name for this method such as checkAndSplitCommand as process might not be a very specific term. See https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well